src/App.tsx

```diff
@@ -9,14 +9,36 @@ interface Todo {
 }
 
 function App() {
-  const [todos, setTodos] = useState<Todo[]>([])
+  const loadTodosFromLocalStorage = (): Todo[] => {
+    const storedTodos = localStorage.getItem('todos')
+    if (storedTodos) {
+      return JSON.parse(storedTodos).map((todo: Todo) => ({
+        ...todo,
+        dueDate: new Date(todo.dueDate)
+      }))
+    }
+    return []
+  }
+
+  const [todos, setTodos] = useState<Todo[]>(loadTodosFromLocalStorage())
   const [newTodo, setNewTodo] = useState('')
   const [dueDate, setDueDate] = useState<string>(new Date().toISOString().split('T')[0])
 
+  const saveTodosToLocalStorage = (todos: Todo[]) => {
+    localStorage.setItem('todos', JSON.stringify(todos))
+  }
+
+  useEffect(() => {
+    if (todos.length > 0) {
```

Comment:
Does this mean if we remove all our to dos, it won't delete the last one in local storage?